### DISCUSSION TITLE
namespaces seem like they'd be better represented by a classification field

### DISCUSSION THREAD

**Discussion by geekosaur** [flora-pm/flora-server#533] [Feature Requests]
The problem with namespaces as implemented is that you need to know which namespace to search in. Since they appear to be mutually exclusive anyway, I am wondering if they would be better represented as a new field "Classification: boot library | core library | general/Hackage | Cardano | …". The namespace seems like it'd be better used to specify possibly overlapping (sub)sets: Hackage, various Stackage snapshots, etc.

**Comment by geekosaur** [comment]
(copied from matrix, roughly)
This also raises the question, to which I don't know the answer, of whether "Cardano" would be a classification or a namespace. I don't even know whether the Cardano packages are on Hackage. (ETA: I am informed they're not on Hackage, which suggests to me it should be a namespace.)

**Comment by tchoutri** [comment]
To clarify the data model, no the namespaces that refer to package repositories won't be turned into 2nd-class membership markers. However the `@haskell` namespace experiment will certainly come to an end.
